api/middleware.go.tmpl

```diff
@@ -52,8 +52,9 @@ func TokenMiddleware(psk []byte, public map[string]string, h http.Handler) http.
 			log.Infof("Authenticating token for protected URL '%s'", r.URL)
 
 			htoken := r.Header.Get("X-Auth-Token")
-			if err := bcrypt.CompareHashAndPassword([]byte(htoken), psk); err != nil {
-				log.Warnf("Unable to authenticate session for '%s' with '%s'", r.URL, htoken)
+			if err := bcrypt.CompareHashAndPassword(psk, []byte(htoken)); err != nil {
```

Comment:
I think this is backwards:

https://godoc.org/golang.org/x/crypto/bcrypt#CompareHashAndPassword

`func CompareHashAndPassword(hashedPassword, password []byte) error`

The hashed password is what comes in via the header.  The plaintext password is what's in the configuration for the API.

Reply:
That explains it.  I think we could probably document the direction better where the bcrypt is supposed to occur - in the config file on the daemon or on the client - unless I missed that or misread it.

Reply:
I was holding the uuid bcrypted in the config file of the daemon lol.

Reply:
ah - that didn't change.  the config file remains the same as it was before.